projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.tsx

```diff
@@ -0,0 +1,207 @@
+            label: this.label,
+            required: this.required
+        };
+        return (
+            <Fragment>
+                {getTagLabel(labelTagParams)}
+                <input
+                    class={getErrorClass(this.status.dotValid)}
+                    disabled={this.disabled || null}
+                    id={this.name}
+                    name="key"
+                    onInput={(event: Event) => this.setValue(event)}
+                    placeholder={this.keyPlaceholder}
+                    type="text"
+                    value={this.fieldInput.key}
+                />
+                <input
+                    class={getErrorClass(this.status.dotValid)}
+                    disabled={this.disabled || null}
+                    name="value"
+                    onInput={(event: Event) => this.setValue(event)}
+                    placeholder={this.valuePlaceholder}
+                    type="text"
+                    value={this.fieldInput.value}
+                />
+                <button
+                    type="button"
+                    disabled={this.disabled || null}
+                    onClick={() => this.addKey()}
+                >
+                    {this.saveBtnLabel}
+                </button>
+                {this.keyValues.length ? (
+                    <key-value-table keyValues={this.keyValues} disabled={this.disabled} />
+                ) : (
+                    ''
+                )}
+                {getTagHint(this.hint)}
+                {getTagError(this.showErrorMessage(), this.getErrorMessage())}
+            </Fragment>
+        );
+    }
+
+    private setInitialValue(): void {
+        this.keyValues = this.value
+            ? this.value
+                  .split(',')
+                  .filter((item) => item.length > 0)
+                  .map((item) => {
+                      const [key, value] = item.split('|');
+                      return { key, value };
+                  })
+            : [];
+        this.status = getOriginalStatus(this.isValid());
+    }
+
+    private isValid(): boolean {
+        return !(this.required && !this.keyValues.length);
+    }
+
+    private showErrorMessage(): boolean {
+        return this.getErrorMessage() && !this.status.dotPristine;
+    }
+
+    private getErrorMessage(): string {
+        return this.isValid() ? '' : this.requiredmessage;
+    }
+
+    private setValue(event): void {
+        this.fieldInput[event.target.name] = event.target.value.toString();
+    }
+
+    private refreshStatus(): void {
+        this.status = updateStatus(this.status, {
+            dotTouched: true,
+            dotPristine: false,
+            dotValid: this.isValid()
+        });
+    }
+
+    private emitStatusChange(): void {
+        this.statusChange.emit({
+            name: this.name,
+            status: this.status
+        });
+    }
+
+    private emitValueChange(): void {
+        const returnedValue = this.keyValues
```

Comment:
This is repeated in the `dot-form`, why not move this to a key-value specific util file?